server/rest/src/test/java/org/infinispan/rest/RestOperationsTest.java

```diff
@@ -440,4 +440,49 @@ public void testConfigConverter() throws Exception {
       assertEquals(2, jsonNode.findValue("string-keyed-jdbc-store").size());
    }
 
+   @Test
+   public void testWeakCounter() throws Exception {
+      String url = String.format("http://localhost:%d/rest/v2/counters/weak", restServer().getPort());
+      ContentResponse response = client.newRequest(url)
+            .method(HttpMethod.DELETE).send();
+      ResponseAssertion.assertThat(response).isOk();
+      response = client.newRequest(url)
+            .method(HttpMethod.GET).send();
+      ResponseAssertion.assertThat(response).isOk();
+      assertEquals(0, Long.parseLong(response.getContentAsString()));
+      response = client.newRequest(url)
+            .method(HttpMethod.POST).content(new StringContentProvider("10")).send();
+      ResponseAssertion.assertThat(response).isOk();
+      for(int i = 0; i < 10; i++) {
+         response = client.newRequest(url)
+               .method(HttpMethod.GET).send();
+         ResponseAssertion.assertThat(response).isOk();
+         long value = Long.parseLong(response.getContentAsString());
+         if (value == 10)
+            return;
+         assertEquals(0, value);
+         Thread.sleep(100);
+      }
```

Comment:
some exception telling the counter's didn't change in time? Don't we have access to `eventually()` methods here?